causalml/inference/tree/causal/causaltree.py

```diff
@@ -234,6 +234,7 @@ def fit_predict(
             )
             te_lower = np.percentile(te_bootstraps, (self.alpha / 2) * 100, axis=0)
             te_upper = np.percentile(te_bootstraps, (1 - self.alpha / 2) * 100, axis=0)
+            te = te_bootstraps.mean(axis=0)
```

Comment:
Here, it's correct to use the original estimate `te` instead of its bootstrapped mean. See the discussion here: https://stats.stackexchange.com/a/71365

Reply:
Hi, @jeongyoonlee,

Thanks for the link!
I got quite interesting results of `test_fit_predict` function in `test_causal_trees.py` with the use of synthetic regression data and the original estimate `te`.
For a randomized trial (mode=2), test shows that `(te_lower <= te).all() and (te_upper >= te).all()` condition doesn't hold.
For difficult nuisance components and an easy treatment effect (mode=1) the above expression is true.

Reply:
Hmm, that's interesting. It seems bootstrapping is not stable for `mode=2`. Should we increase the treatment effect size or the number of bootstraps for the sake of the test?

Reply:
I think we could discuss this in a new issue. I'll continue bootstrap tests a bit later.